federatedml/feature/OHE_with_alignment/OHE_alignment_guest.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# import ptvsd 
+
+# ptvsd.enable_attach(address =('0.0.0.0',6758))
+# ptvsd.wait_for_attach()
+#
+#  added by jsweng
+#  alignemnet guest
+
+import numpy as np
+
+from arch.api.utils import log_utils
+from federatedml.framework.homo.procedure import aggregator
+from federatedml.framework.homo.procedure import paillier_cipher
+from federatedml.feature.OHE_with_alignment.OHE_alignment_base import OHEAlignmentBase
+from federatedml.util import consts
+import functools
+import ast
+from collections import defaultdict
+
+
+LOGGER = log_utils.getLogger()
+
+
+class OHEAlignmentGuest(OHEAlignmentBase):
+    def __init__(self):
+        super(OHEAlignmentGuest, self).__init__()
+        self.role = consts.GUEST
+        # self.aggregator = aggregator.Guest()
+        
+    def _init_model(self, params):
+        super()._init_model(params)
+
+    def fit(self, data_instances):
+
+        """This function allows for one-hot-encoding of the 
+        columns with or without alignment with the other parties
+        in the federated learning.
+
+        Args:
+            data_instances: data the guest has access to
+
+        Returns:
+            if alignment is on, then the one-hot-encoding data_instances are done with
+            alignment with parties involved in federated learning else,
+            the data is one-hot-encoded independently
+
+        """
+
+
+        self.init_schema(data_instances)
+        # ptvsd.break_into_debugger()
+        self._init_params(data_instances)
+        ori_header = self.inner_param.header.copy() # keep a copy of original header
+
+        #obtain the individual column headers with their values
+        f1 = functools.partial(self.record_new_header,
+                               inner_param=self.inner_param)
+        self.col_maps = data_instances.mapPartitions(f1).reduce(self.merge_col_maps)
+        col_maps = {}
+        for col_name, pair_obj in self.col_maps.items():
+            values = [str(x) for x in pair_obj.values]            
+            col_maps[col_name] = values
+        
+        LOGGER.debug("**AIFEL** new col_maps is: {}".format(col_maps))
+
+        #Feature alignment is executed in a 'switch' manner
+        if self.need_alignment:
+
+            #Send col_maps to arbiter
+            self.transfer_variable.guest_columns.remote("{}".format(col_maps), role = consts.ARBITER, idx = -1) # send guest size to arbiter
```

Comment:
FATE can transfer dict directly. Thus, I can't get why you transfer to string before remote.

Reply:
refactor and removed this file